api/controllers/profile-cotroller.js

```diff
@@ -14,7 +14,7 @@ const updateProfile = async (req, res) => {
 const getProfile = async (req, res) => {
     try{
         const { id } = req.body;
-        const {password, ...user} = await User.findById(id);
+        const {...user} = await User.findById(id);
```

Comment:
should detach password from user object for response, so it's correct code 
const {password, ...user} = await User.findById(id);

Reply:
If we use this we get such an answer:
`{
    "$__": {
        "strictMode": true,
        "selected": {},
        "getters": {},
        "_id": "5c5c56ec456a6b69348e120b",
        "wasPopulated": false,
        "activePaths": {
            "paths": {
                "password": "init",
                "email": "init",
                "username": "init",
                "_id": "init",
                "__v": "init"
            },
            "states": {
                "ignore": {},
                "default": {},
                "init": {
                    "_id": true,
                    "username": true,
                    "email": true,
                    "password": true,
                    "__v": true
                },
                "modify": {},
                "require": {}
            },
            "stateNames": [
                "require",
                "modify",
                "init",
                "default",
                "ignore"
            ]
        },
        "pathsToScopes": {},
        "cachedRequired": {},
        "emitter": {
            "_events": {},
            "_eventsCount": 0,
            "_maxListeners": 0
        },
        "$options": {
            "skipId": true,
            "isNew": false,
            "willInit": true
        }
    },
    "isNew": false,
    "_doc": {
        "_id": "5c5c56ec456a6b69348e120b",
        "username": "testing",
        "email": "testing@gmail.com",
        "password": "$2a$10$QQsx8C00qjz1qJboZlkQnOz6WopYR0sZoSCOYsbrVc05kmCvKmeaG",
        "__v": 0
    },
    "$init": true
}`

Reply:
If we don't need password we can use something like this:
`const user = await User.findById(id).select("-password");
`